workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      test-grammar:
+        type: boolean
+        default: true
+      test-rust:
+        type: boolean
+        default: false
+
+jobs:
+  grammar:
+    name: Test grammar
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14]
+    if: ${{inputs.test-node}}
+    steps:
+      - name: Checkout repository
```

Comment:
Nitpick: I dislike giving a name for checking out a repo since it's so common it's obvious what happens.

Reply:
But then you have just one unnamed workflow.

Reply:
You mean unnamed step? Eh, I guess. Anyway it's not really important, let's drop it.